FilePersistence/src/version_control/Merge.cpp

```diff
@@ -43,6 +43,11 @@ namespace FilePersistence {
 
 bool Merge::commit(const Signature& author, const Signature& committer, const QString& message)
 {
+	if (stage_ == Stage::Committed)
```

Comment:
Why did you bring in this commit again? didn't you say it's not needed? Please, use you git skills to clean your branch such that neither this nor the next commit are part of the PR. You can use an interactive rebase for this (google it).